esmvaltool/diag_scripts/weathertyping/wt_utils.py

```diff
@@ -0,0 +1,1675 @@
+        return m_list
+
+    merged_tuples = find_intersection(s)
+    mapping_dict = turn_list_to_mapping_dict(merged_tuples)
+
+    return mapping_dict
+
+
+def write_mapping_dict(work_dir: str, dataset: str, mapping_dict: dict):
+    mapping_dict_reformat = convert_dict(mapping_dict)
+
+    with open(f'{work_dir}/wt_mapping_dict_{dataset}.json',
+              'w',
+              encoding='utf-8') as file:
+        json.dump(mapping_dict_reformat, file)
+
+
+def calc_slwt_obs(cfg: dict, lwt: np.array, cube: iris.cube.Cube,
+                  dataset: str, ancestors: list) -> np.array:
+    """Calculate simplified weathertypes for observation datasets based on
+    precipitation patterns over specified area.
+
+    Args:
+        cfg (dict): Configuration dictionary from recipe
+        lwt (np.array): Array of Lamb WT
+        cube (iris.cube.Cube): preprocessor cube to keep time coordinate
+        dataset (str): Name of dataset
+        correlation_thresold (float): correlation_threshold
+        rmse_threshold (float): rsme_threshold
+        ancestors (list): list of ancestors
+
+    Returns:
+        np.array: _description_
+    """
+
+    logger.info('Calculating simplified Lamb Weathertypes for %s', dataset)
+
+    work_dir = cfg.get('work_dir')
+    correlation_threshold = cfg.get('correlation_threshold')
+    rmse_threshold = cfg.get('rmse_threshold')
+    tcoord = cube.coord('time')
+
+    wt_data_prcp = []
+    for wt in range(1, 28):
+        target_indices = np.where(lwt == wt)
+        if len(target_indices[0]) < 1:
+            logger.info(
+                'calc_slwt_obs - CAUTION: Skipped wt %s \
+                for dataset %s!', wt, dataset)
+            continue
+        dates = [
+            tcoord.units.num2date(tcoord.points[i]) for i in target_indices
+        ]
+        if dataset == 'E-OBS':
+            extracted_cube = cube[target_indices]
+        else:
+            extracted_cube = cube.extract(
+                iris.Constraint(time=lambda t: t.point in dates[0]))
+        wt_cube_mean = extracted_cube.collapsed('time', iris.analysis.MEAN)
+        wt_data_prcp.append(wt_cube_mean.data.compressed())
+    selected_pairs = process_prcp_mean(cfg, wt_data_prcp,
+                                       correlation_threshold,
+                                       rmse_threshold, dataset)
+
+    with open(f'{work_dir}/wt_selected_pairs_{dataset}.json',
+              'w',
+              encoding='utf-8') as file:
+        json.dump(selected_pairs, file)
+
+    mapping_dict = get_mapping_dict(selected_pairs)
+
+    write_mapping_dict(work_dir, dataset, mapping_dict)
+
+    provenance_record = get_provenance_record('Lamb Weathertypes',
+                                              ancestors,
+                                              ['Lamb Weathertypes'],
+                                              False, False)
+
+    log_provenance(f'{dataset}_wt_prov', cfg, provenance_record)
```

Comment:
All your calls to log_provenance are only giving filenames without the paths, resulting in the provenance files being created in the folder where I start the tool, not together with the files.